android/src/main/java/com/reactlibrary/securekeystore/RNSecureKeyStoreModule.java

```diff
@@ -50,104 +52,163 @@ public String getName() {
+        encryptRsaPlainText(publicKey, secretKey.getEncoded()));
+
+      Log.i(Constants.TAG, "created new symmetric keys for alias");
+      return secretKey;
     }
   }
 
+  private void setCipherText(String alias, String input) throws GeneralSecurityException, IOException {
+    Storage.writeValues(getContext(), Constants.SKS_DATA_FILENAME + alias,
+      encryptAesPlainText(getOrCreateSecretKey(alias), input));
+  }
+
   @ReactMethod
-  public void remove(String alias, Promise promise) {
+  public void get(String alias, Promise promise) {
     try {
-      KeyStorage.resetValues(getContext(), alias);
-      Log.i(Constants.TAG, "key removed successfully");
-      promise.resolve("key removed successfully");
-
+      promise.resolve(getPlainText(alias));
     } catch (Exception e) {
+      e.printStackTrace();
       Log.e(Constants.TAG, "Exception: " + e.getMessage());
-      promise.reject("{\"code\":6,\"api-level\":" + Build.VERSION.SDK_INT + ",\"message\":" + e.getMessage() + "}");
+      promise.reject("{\"code\":1,\"api-level\":" + Build.VERSION.SDK_INT + ",\"message\":" + e.getMessage() + "}");
     }
   }
 
+  private PrivateKey getPrivateKey(String alias) throws GeneralSecurityException, IOException {
+    KeyStore keyStore = KeyStore.getInstance(getKeyStore());
+    keyStore.load(null);
+    return (PrivateKey) keyStore.getKey(alias, null);
+  }
+
+  private byte[] decryptRsaCipherText(PrivateKey privateKey, byte[] cipherTextBytes) throws GeneralSecurityException, IOException {
+    Cipher cipher = Cipher.getInstance(Constants.RSA_ALGORITHM);
+    cipher.init(Cipher.DECRYPT_MODE, privateKey);
+    return decryptCipherText(cipher, cipherTextBytes);
+  }
+
+  private byte[] decryptAesCipherText(SecretKey secretKey, byte[] cipherTextBytes) throws GeneralSecurityException, IOException {
+    Cipher cipher = Cipher.getInstance(Constants.AES_ALGORITHM);
+    cipher.init(Cipher.DECRYPT_MODE, secretKey);
+    return decryptCipherText(cipher, cipherTextBytes);
+  }
+
+  private byte[] decryptCipherText(Cipher cipher, byte[] cipherTextBytes) throws IOException {
+    ByteArrayInputStream bais = new ByteArrayInputStream(cipherTextBytes);
+    CipherInputStream cipherInputStream = new CipherInputStream(bais, cipher);
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    byte[] buffer = new byte[256];
+    int bytesRead = cipherInputStream.read(buffer);
+    while (bytesRead != -1) {
+      baos.write(buffer, 0, bytesRead);
+      bytesRead = cipherInputStream.read(buffer);
+    }
+    return baos.toByteArray();
+  }
+
+  private SecretKey getSymmetricKey(String alias) throws GeneralSecurityException, IOException {
+    byte[] cipherTextBytes = Storage.readValues(getContext(), Constants.SKS_KEY_FILENAME + alias);
+
+    //TODO: develop only
+    Log.d(Constants.TAG, "reading secret key: " + new String(decryptRsaCipherText(getPrivateKey(alias), cipherTextBytes), "UTF-8"));
```

Comment:
Wow, I was meant to remove this. Will push a PR now; logging the secret key isn't a good idea

Reply:
@tombailey Yeah, right. I also overlooked it ;)